packages/ui-react/src/components/Sidebar/Sidebar.tsx

```diff
@@ -16,29 +16,44 @@ type SidebarProps = {
 
 const Sidebar: React.FC<SidebarProps> = ({ isOpen, onClose, children }) => {
   const { t } = useTranslation('menu');
+  const lastFocusedElementRef = useRef<HTMLElement | null>(null);
+  const sidebarRef = useRef<HTMLDivElement>() as React.MutableRefObject<HTMLDivElement>;
+  const htmlAttributes = { inert: !isOpen ? '' : undefined }; // inert is not yet officially supported in react. see: https://github.com/facebook/react/pull/24730
+
+  useEffect(() => {
+    if (isOpen) {
+      lastFocusedElementRef.current = document.activeElement as HTMLElement;
+      sidebarRef.current.querySelectorAll('a')[0]?.focus({ preventScroll: true });
+    } else {
+      lastFocusedElementRef.current?.focus({ preventScroll: true });
+    }
+  }, [isOpen]);
 
   return (
     <Fragment>
       <div
-        className={classNames(styles.backdrop, {
-          [styles.visible]: isOpen,
-        })}
-        onClick={onClose}
-      />
-      <div
+        ref={sidebarRef}
         className={classNames(styles.sidebar, {
           [styles.open]: isOpen,
         })}
+        id="sidebar"
+        {...htmlAttributes}
       >
         <div className={styles.heading}>
-          <IconButton onClick={onClose} aria-label={t('close_menu')} tabIndex={isOpen ? 0 : -1}>
+          <IconButton onClick={onClose} aria-label={t('close_menu')}>
             <Icon icon={Close} />
           </IconButton>
         </div>
         <nav className={styles.group} onClick={onClose}>
           {children}
         </nav>
       </div>
+      <div
+        className={classNames(styles.backdrop, {
+          [styles.visible]: isOpen,
+        })}
+        onClick={onClose}
+      />
```

Comment:
There is honestly not a significant reason to do that. Also it doesn't matter (much) because z-index fixes the layering. I reverted this change none the less.